lib/compress/zstd_opt.c

```diff
@@ -399,7 +400,7 @@ static U32 ZSTD_insertBt1(
     assert(ip <= iend-8);   /* required for h calculation */
     hashTable[h] = current;   /* Update Hash Table */
 
-    while (nbCompares-- && (matchIndex > windowLow)) {
+    while (nbCompares-- && (matchIndex >= matchLow)) {
```

Comment:
Why change this?

Reply:
This allows us to match against the first position in the window, which was previously not possible. A separate limit variable needs to be introduced so we can avoid trying to match against index 0, which has a lot of spurious matches pointing at it.